### PR TITLE
fix(route2): infer mime-type of fixtures based on extension, serve binary resources correctly

### LINKED ISSUE #8623: route2 stub for img elements using Fixtures returns text/plain as content type
### Current behavior:
route2 when configured with image fixtures returns content text/plain.

image

### Desired behavior:
route2 configured with fixtures should return images with correct content type.

### Test code to reproduce
Here is a repo with a set of tests.
https://github.com/TonyHernandezAtMS/cypress-test-tiny/
I have found that when I first run this test in Chrome, the first test ends up in a loop requesting the image over and over. If I perform a Empty Cache and Hard Reload in the browser then the tests will progress through all of them.

### Versions
5.2.0, Windows, all browsers

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #8623 

### User facing changelog

- Fixed an issue where routes that stub fixtures for binary resources (including images) made with `cy.route2` would not serve the correct mime-type and would have base64-encoded content.

### Additional details

- existing fixture implementation base64's binary resources by default - obviously this is wrong for cy.route2's use case
- also, removed snapshotting from req/res logs, since it was having an exponential logging effect due to #8679 

### How has the user experience changed?

- can now easily stub image responses

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts [hunk 1]
```diff
@@ -638,6 +638,15 @@ describe('network stubbing', function () {
 
         cy.contains('#result', '""').should('be.visible')
       })
+
+      // @see https://github.com/cypress-io/cypress/issues/8623
+      it('works with images', function () {
```
Comment (bahmutov) [inline packages/driver/cypress/integration/commands/net_stubbing_spec.ts:643]: ha, such a good test

File: packages/net-stubbing/lib/server/util.ts [hunk 2]
```diff
@@ -82,25 +83,28 @@ const caseInsensitiveGet = function (obj, lowercaseProperty) {
...
-export async function setBodyFromFixture (getFixtureFn: GetFixtureFn, staticResponse: BackendStaticResponse) {
+export async function setResponseFromFixture (getFixtureFn: GetFixtureFn, staticResponse: BackendStaticResponse) {
   const { fixture } = staticResponse
 
   if (!fixture) {
     return
   }
 
-  const data = await getFixtureFn(fixture.filePath, { encoding: fixture.encoding })
+  const data = await getFixtureFn(fixture.filePath, { encoding: fixture.encoding || null })
```
Comment (bahmutov) [inline packages/net-stubbing/lib/server/util.ts:93]: so nice, the type is `| null` and code is `|| null`

File: packages/net-stubbing/lib/server/util.ts [hunk 3]
```diff
@@ -82,25 +83,28 @@ const caseInsensitiveGet = function (obj, lowercaseProperty) {
...
+    const mimeType = MimeTypes.lookup(fixture.filePath) || parseContentType(data)
+
+    _.set(staticResponse, 'headers.content-type', mimeType)
   }
 
   function getBody (): string {
     // NOTE: for backwards compatibility with cy.route
     if (data === null) {
-      return ''
+      return JSON.stringify('')
```
Comment (bahmutov) [inline packages/net-stubbing/lib/server/util.ts:107]: oh my

File: packages/net-stubbing/lib/server/util.ts (hunk 2 above)
Comment (flotwig) [reply packages/net-stubbing/lib/server/util.ts:93]: yep, to maintain compatibility with the cy.fixture code...

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts (hunk 1 above)
Comment (flotwig) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts:643]: thanks :)

**Comment by bahmutov** [comment]
Still not passing https://github.com/cypress-io/cypress/issues/8934

**Comment by flotwig** [comment]
@bahmutov that seems to be a separate issue, this issue is about stubbing binary resources, so that can be tackled separately

**Comment by bahmutov** [comment]
maybe it is a different issue, but seems the binary is treated differently when waiting for resources, so thought it would be useful to link

**Comment by flotwig** [comment]
absolutely, thanks for creating an issue, it still needs to be addressed for sure

**Review by bahmutov — APPROVED** [review]
It is working now with images, opened https://github.com/cypress-io/cypress-example-recipes/pull/579 to verify it

**Pull request merged**